### PR TITLE
Make tinymce into a package.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Context

Make tinymce into a package.

## Relevant Technical Choices

Also moved the tests from __tests__ to test directories. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -183,7 +184,6 @@
     "percy": "^0.28.6",
     "prettier": "^2.2.1",
     "puppeteer": "^5.3.1",
-    "react-test-renderer": "^16.14.0",
```
Comment (swissspidy) [inline package.json]: Why was this removed?

File: packages/tinymce/src/utils/test/index.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ jest.mock('../../utils/globals', () => ({
   },
 }));
 
-import * as Utils from '../../utils';
+import * as Utils from '../index';
```
Comment (swissspidy) [inline packages/tinymce/src/utils/test/index.js]:
```suggestion
import * as Utils from '../';
```

File: packages/tinymce/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# TinyMCE
+
+Adds a button to WordPress's TinyMCE interface.
```
Comment (swissspidy) [inline packages/tinymce/README.md]: Let's call the package `@web-stories-wp/tinymce-button`. Just `tinymce` is too generic.

File: package.json (hunk 1 above)
Comment (swissspidy) [reply package.json]: This is used by Testing Library and should definitely stay in the main `package.json` file.

File: packages/tinymce/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  "name": "@web-stories-wp/tinymce",
+  "private": true,
+  "engines": {
+    "node": ">= 15",
+    "npm": "~7"
+  },
+  "exports": {
+    "default": "./src/web-stories.js"
+  },
+  "main": "./src/web-stories.js",
```
Comment (swissspidy) [inline packages/tinymce/package.json]: Let's rename the `./src/web-stories.js` file to `./src/index.js` for consistency with other packages' entry points.

File: packages/tinymce/package.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    "prop-types": "^15.7.2",
+    "@wordpress/compose": "^3.23.0",
+    "@wordpress/components": "^12.0.7",
+    "@wordpress/data": "^4.26.0",
+    "@wordpress/element": "^2.19.0",
+    "@wordpress/i18n": "^3.17.0"
+  },
+  "devDependencies": {
+    "react": "^16.14.0",
+    "react-test-renderer": "^16.14.0"
```
Comment (swissspidy) [inline packages/tinymce/package.json]: This dev dependency should stay in the main `package.json`

File: packages/tinymce/package.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "dependencies": {
+    "prop-types": "^15.7.2",
+    "@wordpress/compose": "^3.23.0",
+    "@wordpress/components": "^12.0.7",
+    "@wordpress/data": "^4.26.0",
+    "@wordpress/element": "^2.19.0",
+    "@wordpress/i18n": "^3.17.0"
+  },
+  "devDependencies": {
+    "react": "^16.14.0",
```
Comment (swissspidy) [inline packages/tinymce/package.json]: Why was this added and not using `@wordpress/element`? And why not in `dependencies`?

Comment (spacedmonkey) [reply packages/tinymce/package.json]: React is just used in the tests. So i dont think it is an issue.

Comment (swissspidy) [reply packages/tinymce/package.json]: It becomes an issue if the version of React in `react` is different from the version provided by `@wordpress/element`.

All code - tests and source - should use the same version of the same package, in this case  only `@wordpress/element`

File: package.json [hunk 7]
```diff
@@ -46,6 +46,7 @@
     "@web-stories-wp/stories-lightbox": "*",
     "@web-stories-wp/templates": "*",
     "@web-stories-wp/text-sets": "*",
+    "@web-stories-wp/tinymce-buttons": "*",
```
Comment (swissspidy) [inline package.json]: Why plural `buttons`? I thought it's only 1 button?

Comment (spacedmonkey) [reply package.json]: Oh, read your feedback as buttons. May we have other buttons in the future?

File: packages/tinymce/package.json (hunk 6 above)
Comment (spacedmonkey) [reply packages/tinymce/package.json]: Seems like it wasn't need anymore. Removed in https://github.com/google/web-stories-wp/pull/6578/commits/d333406a7a91a0990c55ce938f6021bf8373d889

File: package.json (hunk 7 above)
Comment (swissspidy) [reply package.json]: I doubt it. Let's keep it singular.

File: webpack.config.cjs [hunk 8]
```diff
@@ -343,7 +343,7 @@ const widgetScript = {
 const storiesMCEButton = {
   ...sharedConfig,
   entry: {
-    'web-stories-button': './assets/src/tinymce/web-stories.js',
+    'web-stories-button': './packages/tinymce/src/web-stories.js',
```
Comment (swissspidy) [inline webpack.config.cjs]:
```suggestion
    'web-stories-button': './packages/tinymce-button/src/index.js',
```

Comment (spacedmonkey) [reply webpack.config.cjs]: I was fixed this myself.

**Review by swissspidy — APPROVED** [review]
